### ISSUE TITLE
[BUG] - TimeSet event issue from NgxMatTimepickerComponent

### ISSUE THREAD

**Issue by ManuBucaciuc** [tonysamperi/ngx-mat-timepicker#43]
**Describe the bug**
I'm using inside my app the time-picker widget set to 12h format . And I'm trying to get use of (timeSet) event but it doesn't work accordingly. Every time the event is triggered, the emitted value is always set to 'AM' even if I set it to be 'PM'.

**Screenshots**
![image](https://user-images.githubusercontent.com/26401523/202746199-4f4f8ab7-d2ad-43c4-b6de-9de23ee69fd1.png)
![image](https://user-images.githubusercontent.com/26401523/202746304-02c63f0c-63ae-44a1-8266-5770226ed560.png)

**Desktop (if execution):**
 - Browser:  Chrome

** Versions
 - ngx-mat-timepicker version: ^14.0.5
 - @angular/core: ^14.2.6
 - @angular/material: ^14.2.5
 - typescript: ~4.8.4

**Comment by LayZeeDK** [comment]
Can you set up a repo or StackBlitz demonstrating the issue, @ManuBucaciuc?

**Comment by tonysamperi** [comment]
I guess there's something wrong in your code. I tried this and it logs correctly.
As LayZeeDK suggest, you should create a Blitz to reproduce the issue and post here.
I'll reopen this issue if necessary.

**Comment by ManuBucaciuc** [comment]
@LayZeeDK @tonysamperi you can find the repo which demonstrates the issue exists here https://github.com/ManuBucaciuc/Time-Picker-Issue. Thanks!

**Comment by tonysamperi** [comment]
@ManuBucaciuc mate, the example got to be a playable one, as already said you should put the code on a Blitz (stackblitz.com).

Cheers

**Comment by ManuBucaciuc** [comment]
@tonysamperi and here is the Blitz: https://stackblitz.com/edit/angular-ivy-sfjkm8?file=package.json,src%2Findex.html,src%2Fapp%2Fapp.component.ts,src%2Fapp%2Fapp.component.html,src%2Fapp%2Fapp.module.ts

All the best.

**Comment by tonysamperi** [comment]
@ManuBucaciuc mmm interesting, I made the same test and it worked. I'll investigate further!

**Comment by tonysamperi** [comment]
You're missing the binding on the input.
If you don't link an input, the picker doesn't have the model reference...

![image](https://user-images.githubusercontent.com/5957244/204087793-503769e8-4d17-45ba-888c-2b236afbd2ad.png)

**Comment by ManuBucaciuc** [comment]
@tonysamperi let me give you the full picture. I'm trying to implement a custom component inside my app that gives my clients the possibility of both typing (by adding a pattern validator for my control: `this.control.addValidators(Validators.pattern(RegexPatterns.AM_PM_TIME_FORMAT));` and also using `mask="00:00 SS"`) and time selection using `<ngx-mat-timepicker>` dialog (opens at matSuffix icon click). The thing is, once I bind the ngxMatTimepicker to input, I lose the typing ability (or at least, reading the documentation I didn't find any useful properties that helps me achieving the wanted behavior -- I did use `disableClick` property but with no effect).
Also, by binding both `mask` and `ngxMatTimepicker` to the input I'm facing the following error `More than one custom value accessor matches form control with unspecified name attribute.`.
So, in order to make my wanted behavior work I just had to add one new extra input that is binded to my timePicker and if value changes I just update my control (my custom timepicker component also extends an `ValueAccessorBase` abstract class that implements `ControlValueAccessor`).
Even my solution works as wanted, I'm not 100% percent sure if it's the right approach. Any recommendation will be much appreciated.

Is there any specific reason why we must bind the timePicker to an input in order to work accordingly? While testing without my timePicker being  bind to an input, the value emitted by (timeSet) has the correct hours and minutes and only the AM/PM format is not set accordingly.

**Comment by tonysamperi** [comment]
@ManuBucaciuc this library was born to prevent the users from typing, especially for mobile. If you want your users to type, why bothering adding an extra dependency?  
I don't see the point in having 2 input modes.
ANYWAY
I understand what you're trying to do, but this library simply wasn't born in a way that allows what you're looking for (and btw I didn't design the base architecture, I simply improved some parts).
FINALLY
I spent 30 seconds thinking about your use case and had a very easy workaround.
But since this is not stackoverflow I don't want to take from you the pleasure of finding the solution.

Cheers

**Comment by tonysamperi** [comment]
Hi @ManuBucaciuc 

**TL;DR** you can bypass the issue by adding the "format" parameter on the picker

```
<ngx-mat-timepicker (timeSet)="updateTime($event, 'selectedTimeFreeInput')"
                                    #pickerH></ngx-mat-timepicker>
```

**LONG STORY IF YOU'RE INTERESTED**
I was digging into your problem, because after thinking about it, the error didn't make sense.
And actually the event doesn't work as I thought (it actually works with observables). I assumed that your blitz started working after linking the picker but it actually wasn't that. 
There's a fundamental problem, which is the ability to set options both on the input and on the picker itself.
BUT.
While on the directive you get defaults, you don't get them on the picker. 
And that is a big african pine cone in the arse (this is a quote from some cartoon I saw, don't worry if it doesn't make sense to you 😂).
So I'll need to actually fix this...and then I'm quite inclined to remove all the params from the directive side of the moon.
But for now I'll just find the missing initializer and fix it.

Cheers

**Comment by tonysamperi** [comment]
Fixed in v14.0.6

**Comment by ManuBucaciuc** [comment]
@tonysamperi Thank you mate. For discussions, investigations, solution and fix.

Take care
